File: app/src/main/java/com/jqorz/planewar/frame/GameLauncher.java
package com.jqorz.planewar.frame;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.ImageView;

import com.jqorz.planewar.R;

public class GameLauncher extends Activity implements View.OnClickListener, View.OnLongClickListener {
    private ImageView imgBtn_startGame, imgBtn_exitGame, imgBtn_about, imgBtn_setting;
    private ImageView iv_icon;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.game_launcher);
        initView();
        initEvent();
    }

    private void initView() {
        imgBtn_startGame = (ImageView) findViewById(R.id.imgBtn_startGame);
        imgBtn_exitGame = (ImageView) findViewById(R.id.imgBtn_exitGame);
        imgBtn_about = (ImageView) findViewById(R.id.imgBtn_about);
        imgBtn_setting = (ImageView) findViewById(R.id.imgBtn_Setting);
        iv_icon = (ImageView) findViewById(R.id.iv_icon);
    }

    private void initEvent() {
        imgBtn_startGame.setOnClickListener(this);
        imgBtn_exitGame.setOnClickListener(this);
        imgBtn_about.setOnClickListener(this);
        imgBtn_setting.setOnClickListener(this);
        iv_icon.setOnLongClickListener(this);
    }


    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.imgBtn_startGame:
                this.startActivity(new Intent(this, GamePlaying.class));
                this.finish();
                break;
            case R.id.imgBtn_about:
                this.startActivity(new Intent(this, GameAbout.class));
                break;
            case R.id.imgBtn_Setting:
                this.startActivity(new Intent(this, GameSetting.class));
                break;
            case R.id.imgBtn_exitGame:
                this.finish();
                break;
        }
    }


    @Override
    public boolean onLongClick(View view) {
        if (view.getId() == R.id.iv_icon) {
            Intent intent = new Intent(this, GameCheat.class);
            this.startActivity(intent);
        }
        return false;
    }
}

File: app/src/main/java/com/jqorz/planewar/frame/GameLauncher.kt
package com.jqorz.planewar.frame

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.FragmentActivity
import android.view.View
import android.view.View.OnLongClickListener
import com.jqorz.planewar.R
import com.jqorz.planewar.tools.UserDataManager
import kotlinx.android.synthetic.main.game_launcher.*

class GameLauncher : FragmentActivity(), View.OnClickListener, OnLongClickListener {
    private lateinit var mSettingDialog: GameSettingDialog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.game_launcher)
        initEvent()
    }


    private fun initEvent() {
        imgBtn_startGame!!.setOnClickListener(this)
        imgBtn_exitGame!!.setOnClickListener(this)
        imgBtn_about!!.setOnClickListener(this)
        imgBtn_setting!!.setOnClickListener(this)
        iv_icon!!.setOnLongClickListener(this)

    }


    override fun onClick(view: View) {
        when (view.id) {
            R.id.imgBtn_startGame -> {
                this.startActivity(Intent(this, GamePlaying::class.java))
                finish()
            }
            R.id.imgBtn_about -> this.startActivity(Intent(this, GameAbout::class.java))
            R.id.imgBtn_setting -> {
                showSettingDialog()
            }
            R.id.imgBtn_exitGame -> finish()
        }
    }

    fun showSettingDialog() {
        mSettingDialog = GameSettingDialog(this)
        mSettingDialog.show()
        mSettingDialog.setSwitch(UserDataManager.isOpenMusic(), UserDataManager.isOpenSound())
        mSettingDialog.callback = object : GameSettingDialog.DialogCallback {
            override fun onSwitchSound(open: Boolean) {
                UserDataManager.setOpenSound(open)
            }

            override fun onSwitchMusic(open: Boolean) {
                UserDataManager.setOpenMusic(open)
            }
        }
    }

    override fun onLongClick(view: View): Boolean {
        if (view.id == R.id.iv_icon) {
            GameCheat.start(this)
        }
        return true
    }

    companion object {
        @JvmStatic
        fun start(context: Context) {
            val starter = Intent(context, GameLauncher::class.java)
            context.startActivity(starter)
        }
    }
}